src/datumaro/experimental/export_import.py

```diff
@@ -0,0 +1,432 @@
+        output_path: Path to export to (directory or .zip file)
+        export_images: Whether to export images from callable/path fields
+        as_zip: Whether to package everything as a ZIP file
+
+    Raises:
+        ValueError: If the dataset has transforms (not yet supported)
+    """
+    output_path = Path(output_path)
+
+    # Create temporary directory if using zip, otherwise use output_path directly
+    temp_dir = None
+    if as_zip:
+        if output_path.suffix != ".zip":
+            output_path = output_path.with_suffix(".zip")
+        temp_dir = Path(tempfile.mkdtemp())
+        work_dir = temp_dir
+    else:
+        output_path.mkdir(parents=True, exist_ok=True)
+        work_dir = output_path
+
+    try:
+        # Export DataFrame to Parquet
+        # Filter out Object columns (like callables) as they can't be serialized to Parquet
+        columns_to_export = [col for col, dtype in dataset.df.schema.items() if dtype != pl.Object]
+        parquet_path = work_dir / DATAFRAME_FILE
+        dataset.df.select(columns_to_export).write_parquet(parquet_path)
+
+        # Export images if requested
+        if export_images:
+            images_dir = work_dir / IMAGES_DIR
+            _export_images_from_dataset(dataset, images_dir)
+
+        # Track which columns are Object types (excluded from parquet)
+        object_columns = [col for col, dtype in dataset.df.schema.items() if dtype == pl.Object]
+
+        # Create metadata
+        metadata = {
+            "version": VERSION,
+            "schema": dataset.schema.to_dict(),
+            "object_columns": object_columns,
+        }
+
+        # Write metadata
+        metadata_path = work_dir / METADATA_FILE
+        with open(metadata_path, "w") as f:
+            json.dump(metadata, f, indent=2)
+
+        # Create ZIP if requested
+        if as_zip:
+            with ZipFile(output_path, "w", ZIP_DEFLATED) as zipf:
+                for file_path in work_dir.rglob("*"):
+                    if file_path.is_file():
+                        arcname = file_path.relative_to(work_dir)
+                        zipf.write(file_path, arcname)
+
+    finally:
+        # Clean up temporary directory if used
+        if as_zip and temp_dir is not None and temp_dir.exists():
+            shutil.rmtree(temp_dir)
+
+
+def import_dataset(
+    input_path: Union[str, Path],
+    dtype: Type["DType"] | None = None,
+) -> "Dataset[DType]":
+    """
+    Import a dataset from an exported format.
+
+    Args:
+        input_path: Path to the exported dataset (directory or .zip file)
+        dtype: Optional Sample class to use for the dataset. If None, uses generic Sample.
+
+    Returns:
+        The imported Dataset instance
+
+    Raises:
+        FileNotFoundError: If required files are missing
+        ValueError: If the dataset format is invalid
+    """
+    input_path = Path(input_path)
+
+    # Handle ZIP files
+    if input_path.suffix == ".zip":
```

Comment:
The simplest and most reliable way to check if a file is a valid ZIP file in Python is to use the is_zipfile() function from the built-in zipfile module.

This function checks the magic number (the identifying bytes) at the beginning of the file, not just the file extension, making it a robust check.